### ISSUE TITLE
Nikon Z 9 Camera Support

### ISSUE THREAD

**Issue by peti** [Beep6581/RawTherapee#6408]
I have uploaded three sample NEF files created by the Nikon Z 9 camera to https://raw.pixls.us/:

- `nikon-z9-raw-lossless.nef`
- `nikon-z9-raw-compressed-high-quality.nef`
- `nikon-z9-raw-compressed-low-quality.nef`

The Z 9 does not support uncompressed RAW. The camera knows only "RAW L" with different levels of compression. The compression modes are:

- "loss-less",
- "high efficiency*", and 
- "high efficiency",

where the star variant ("`*`") denotes higher quality a.k.a. less efficient compression.

The images show an X-Rite Color Checker Passport.

I have not touched the files except renaming them for clarity; these come straight out of the camera.

**Comment by ValZapod** [comment]
And now you will have to add picoraw too, that is going to be very hard i imagine.

**Comment by StepanDav** [comment]
The recently released (and seemingly popular) Nikon Z8 is in many ways similar to the Z9.

RawTherapee 5.9 seems to support "lossless" NEF files. However, neither lossy compression level is supported. The files open as black.

 How can the community help bring support for the compressed NEF files into Rawtherapee?

**Comment by Lawrence37** [comment]
A quick scan shows neither LibRaw nor rawspeed can decode "high efficiency" compression either. The first step is reverse-engineering the compression or getting the specifications from Nikon.

**Comment by StepanDav** [comment]
Based on [some survey of Z9 support](https://www.nikoncafe.com/threads/nikon-z9-raw-photo-nraw-video-support-16-mar-2023.331229/), it would appear that the compression within is https://www.intopix.com/tico-raw 

Their codec was previously widely reported as used for N-RAW video, but it looks like it's used for static photos too.

**Comment by kmilos** [comment]
> Based on [some survey of Z9 support](https://www.nikoncafe.com/threads/nikon-z9-raw-photo-nraw-video-support-16-mar-2023.331229/), it would appear that the compression within is https://www.intopix.com/tico-raw

Yay. And?

P.S. [This was already a known fact since the Z 9 launch](https://www.dpreview.com/news/9624409613/nikon-is-licensing-intopix-s-ticoraw-technology-for-its-z9-camera-system). Note the word "licensing" in the title and the phrase "patented compression technology" in the text.

**Comment by kmilos** [comment]
FYI: https://www.libraw.org/comment/6674#comment-6674

**Comment by struffel** [comment]
Just adding a screenshot of the proper raw setting to make things clearer for new Z8 owners coming from Google looking for the right setting (it's in the "Photo Shooting Menu" -> "Raw Recording"):
![image](https://github.com/Beep6581/RawTherapee/assets/31403260/fe12cc88-9e4c-43a6-8ce0-fc8552dcd78a)

**Comment by isenberg** [comment]
According to https://www.nikoncafe.com/threads/nikon-he-he-nraw-support-as-of-12-july-2024.331229 the new patented high efficiency encoding decreases .nef file size by 40% from 55-60 MB to only 33 MB.

NASA's current ISS photos from the space station on http://eol.jsc.nasa.gov  taken with the few Nikon Z9 on board now since about a year are encoded with this when downloading the raw .nef next to the also available .jpg. There it makes sense to use the higher compression as bandwidth is a huge cost factor.

**Comment by isenberg** [comment]
As workaround I used the free https://helpx.adobe.com/camera-raw/using/adobe-dng-converter.html to convert to .dng and it can process a whole directory, but only Windows and macOS, no Linux.

Does anyone know a Linux command line tool to read the high efficiency .nef?

**Comment by purew** [comment]
> Does anyone know a Linux command line tool to read the high efficiency .nef?

The compression is proprietary and needs licensing. You're not likely to find it in any open source tool. 

Have you tried running the Adobe converter on Linux with Wine?

**Comment by isenberg** [comment]
> Have you tried running the Adobe converter on Linux with Wine?

No as I have macOS on the desktop, but that could indeed be a usable workaround for those who only run Linux.

**Comment by Lawrence37** [comment]
Adobe DNG Converter has a command line interface and can be invoked on Linux using Wine. @isenberg I'm curious why you asked if there is a Linux CLI tool if you run macOS? ADC is available for macOS and I assume it can also be used from the command line.

**Comment by bczhc** [comment]
I think i may meet the same issue. RawTherapee is able to read lossless NEF but not the High Efficiency* one. See the screenshot: 

![Image](https://github.com/user-attachments/assets/6b5512fa-c8f6-43a1-8a26-76d613d72bfa)

However, the linux software [Geeqie](https://en.wikipedia.org/wiki/Geeqie) can just show a correct preview of the HE*.

Edit: Camera: Z50 Ⅱ

**Comment by isenberg** [comment]
> RawTherapee is able to read lossless NEF but not the High Efficiency* one.

Yes, that's the same issue. As far I know, the camera default setting is high efficiency compression enabled. At least that's the setting NASA uses on the ISS unfortunately. They release the raw .NEF files a few months after the JPEGs usually where I noticed it.

**Comment by Lawrence37** [comment]
> However, the linux software Geeqie can just show a correct preview of the HE*.

And so can RawTherapee

**Comment by isenberg** [comment]
>> However, the linux software Geeqie can just show a correct preview of the HE*.
> And so can RawTherapee

But only the embedded JPEG-preview image inside the HE-compressed .NEF.

**Comment by RdWing** [comment]
Hello, LibRAW fully supports HE and HE* compressed Nikon RAW files for awhile now. Can we please have it enabled inside RawTherapee?

**Comment by kmilos** [comment]
> Hello, LibRAW fully supports HE and HE* compressed Nikon RAW files for awhile now.

It does not currently.

**Comment by RdWing** [comment]
> > Hello, LibRAW fully supports HE and HE* compressed Nikon RAW files for awhile now.
> 
> It does not currently.

It does, it currently functions in RawDigger and FastRawViewer. [Support was added in v2.09](https://www.fastrawviewer.com/blog/FastRawViewer-2-0-9-Release#:~:text=Camera%2FRAW%20format%20support%0A%0ANikon%20HE%2FHE*%20compression%20support%20(for%20Nikon%20Z8%2C%20Z9%2C%20Zf%2C%20Z6_3%20cameras)).

**Comment by kmilos** [comment]
RawDigger and FastRawViewer are paid for, and not open source products by LibRaw LLC (the commercial entity). LibRaw (the library) is not the same thing, and support for HE has not been added to it and publicly released yet.

**Comment by RdWing** [comment]
Well that is terribly confusing. I understand RawDigger et al are commercial products from libRaw LLC, but they claim to maintain and use LibRaw the library. [And the web page for LibRaw the library says](https://www.libraw.org/supported-cameras#:~:text=if%20our%20RawDigger%20announced%20at%20our%20blog%2C%20https%3A%2F%2Fwww.rawdigger.com%2Fnews%2C%20supports%20the%20camera%20%2F%20format%2C%20that's%20because%20LibRaw%20supports%20it.), "If Rawdigger announces support, it's because LibRaw supports it".
And it sounds like LibRaw does support it, but it has not been pushed to GitHub yet.

**Comment by akontsevich** [comment]
Not sure about LibRaw, however why [PhotoQt v4.9.2](https://gitlab.com/lspies/photoqt/-/releases/v4.9.2) can show everything to me from Nikon Z8 (NEF, HEIF), Darktable opens only HEIF, but RawTherapee can't open anything?! Can provide links to photos if necessary or upload to https://raw.pixls.us/.

**Comment by Lawrence37** [comment]
What exactly is PhotoQt showing? Have you verified that it shows the raw data and not the embedded preview inside the raw file?

**Comment by RdWing** [comment]
> Not sure about LibRaw, however why [PhotoQt v4.9.2](https://gitlab.com/lspies/photoqt/-/releases/v4.9.2) can show everything to me from Nikon Z8 (NEF, HEIF), Darktable opens only HEIF, but RawTherapee can't open anything?! Can provide links to photos if necessary or upload to https://raw.pixls.us/.

I tested this just now, and PhotoQt only shows you the embedded JPEG. It is unable to decode the RAW. If you set PhotoQt not to show the embedded preview file, you'll see the corrupt raw data.

<img width="2105" height="1630" alt="Image" src="https://github.com/user-attachments/assets/a8cfdcdb-71c8-4cad-9149-b73a337dcefb" />

**Comment by isenberg** [comment]
In case anyone like to try the original NASA ISS Nikon Z9 HE lossy compressed NEF 14 bit files. Here a direct link to one:
https://eol.jsc.nasa.gov/SearchPhotos/photo.pl?mission=ISS072&roll=E&frame=410334

To get the raw NEF, go to Download Options on the same page, then the button at "Raw File from camera Request".

That will extract the file from the archive which can take a few minutes and makes it available on this URL:
https://eol.jsc.nasa.gov/OriginalImagery/iss072e410334.NEF

That URL will currently work for a few hours maybe as it caches the result.

**Comment by isenberg** [comment]
I see the raw .NEF files from the Nikon D5 and Z9 on the Artemis 2 mission are compressed, which means they won't open in RawTherapee when they are released within a few months. They are not yet available to the public, only for a few selected individuals apparently, but in the full res JPEG metadata, the NEF compression flag can be seen. The fullres JPEGs with metadata are are available on this link: https://eol.jsc.nasa.gov/SearchPhotos/mrf.pl?scope=both&MRFList=ART002

Darktable hit the same problem: https://github.com/darktable-org/darktable/issues/15873

Artemis 2 metadata:
NASA PHOTO ID				art002e009175.NEF
GMT					2026:04:05 06:34:45
MODEL					NIKON D5 S/N: 3500017
Firmware				Ver.N.10
Image Size				5584x3728
Compression				Nikon NEF Compressed
